### ISSUE TITLE
forget _apply_sparse_duplicate_indices?

### ISSUE THREAD

**Issue by zhangys-lucky** [Qihoo360/tensornet#29]
tensorflow 在apply sparse gradient的时候先对重复的indice做了去重，tensornet是否遗漏了相关的去重工作。
[https://github.com/tensorflow/tensorflow/blob/5681c179eff80bce00e526303950b67b23cad14c/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L1100](https://github.com/tensorflow/tensorflow/blob/5681c179eff80bce00e526303950b67b23cad14c/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L1100)

**Comment by justsmoke** [comment]
感觉tn这一块代码有些问题，optimizer.py没有处理sparse gradient
[https://github.com/Qihoo360/tensornet/blob/master/tensornet/optimizer/optimizer.py#L49](url)
如果用了tf.gather或其他产生sparse gradient的op就会挂
@zhangys-lucky 大神有什么解决方案吗？
